plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -454,7 +455,10 @@ hoistExpr var t = do
     -- See Note [Dependency tracking]
     modifyCurDeps (Set.insert lexName)
     maybeDef <- PIR.lookupTerm annMayInline lexName
-    case maybeDef of
+    let addSpan = case getVarSourceSpan var of
+            Nothing  -> id
+            Just src -> fmap . fmap . addSrcSpan $ spanToSpan src
+    addSpan $ case maybeDef of
```

Comment:
This is wrong, I think. The expression here is the expression we're going to insert at the _use site_, i.e. we see `fib` and this will eventually produce a variable reference to `fib`. _That_ shouldn't have a location based on the definition location of `fib`, the _body_ of `fib` should have that. So I think this needs to go on line 472 instead.

Reply:
472 in the new diff, the call to `compileExpr`, anyway